crates/assists/src/handlers/extract_assignment.rs

```diff
@@ -0,0 +1,288 @@
+use syntax::{
+    ast::{self, edit::AstNodeEdit, make},
+    AstNode,
+};
+use test_utils::mark;
+
+use crate::{
+    assist_context::{AssistContext, Assists},
+    AssistId, AssistKind,
+};
+
+// Assist: extract_assignment
+//
+// Extracts variable assigment to outside an if or match statement.
+//
+// ```
+// fn main() {
+//     let mut foo = 6;
+//
+//     if true {
+//         <|>foo = 5;
+//     } else {
+//         foo = 4;
+//     }
+// }
+// ```
+// ->
+// ```
+// fn main() {
+//     let mut foo = 6;
+//
+//     foo = if true {
+//         5
+//     } else {
+//         4
+//     };
+// }
+// ```
+pub(crate) fn extract_assigment(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let name = ctx.find_node_at_offset::<ast::NameRef>()?.to_string();
```

Comment:
```suggestion
    let name = ctx.find_node_at_offset::<ast::NameRef>()?.as_name();
```
With this and the comment below we can get rid of the string usage and compare by `Name`s I believe. The method there comes from [AsName](https://github.com/rust-analyzer/rust-analyzer/blob/master/crates/hir_expand/src/name.rs#L75)